xmtp/src/conversations.rs

```diff
@@ -647,6 +813,70 @@ mod tests {
         assert_eq!(conversations.len(), 0);
     }
 
+    #[tokio::test]
+    async fn process_messages_happy_path() {
+        init();
+        let (alice_client, bob_client) = gen_two_test_clients().await;
+
+        let bob_address = bob_client.account.contact().wallet_address;
+
+        let a_convos = Conversations::new(&alice_client);
+        let b_convos = Conversations::new(&bob_client);
+
+        let a_to_b = a_convos
+            .new_secret_conversation(bob_address.clone())
+            .unwrap();
+
+        // Send First Message
+
+        a_to_b.send_message("Hi").unwrap();
+        alice_client
+            .refresh_user_installations(&bob_address)
+            .await
+            .unwrap();
+        a_convos.process_outbound_messages().await.unwrap();
+        a_convos.publish_outbound_payloads().await.unwrap();
+        b_convos.receive().unwrap();
+
+        let bob_messages = bob_client
+            .store
+            .get_stored_messages(&mut bob_client.store.conn().unwrap())
+            .unwrap();
+
+        assert_eq!(bob_messages.len(), 1);
+
+        {
+            let alice_messages = alice_client
+                .store
+                .get_stored_messages(&mut alice_client.store.conn().unwrap())
+                .unwrap();
+            assert_eq!(alice_messages.len(), 1);
+        }
+
+        // Reply
+        let b_to_a = b_convos
+            .new_secret_conversation(bob_address.clone())
+            .unwrap();
+
+        b_to_a.send_message("Reply").unwrap();
+        bob_client
+            .refresh_user_installations(&bob_address)
+            .await
+            .unwrap();
+        b_convos.process_outbound_messages().await.unwrap();
+        b_convos.publish_outbound_payloads().await.unwrap();
+
+        a_convos.receive().unwrap();
+
+        let _alice_messages = alice_client
+            .store
+            .get_stored_messages(&mut alice_client.store.conn().unwrap())
+            .unwrap();
+
+        // TODO: This is currently failing with a NoSession error for unknown reasons
+        // assert_eq!(alice_messages.len(), 2);
```

Comment:
This is a follow up piece of work. The sending side is not finding a session when replying to a message.